api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerUnlockRecipes.java

```diff
@@ -0,0 +1,331 @@
+                                        boolean smeltingRecipeBookFilterActive, int elements, Optional<String[]> recipeIdsModern, OptionalInt elementsInit,
+                                        Optional<String[]> recipeIdsModernInit) {
+    super(PacketType.Play.Server.UNLOCK_RECIPES);
+    this.action = action;
+    this.craftingRecipeBookOpen = craftingRecipeBookOpen;
+    this.craftingRecipeBookFilterActive = craftingRecipeBookFilterActive;
+    this.smeltingRecipeBookOpen = smeltingRecipeBookOpen;
+    this.smeltingRecipeBookFilterActive = smeltingRecipeBookFilterActive;
+    this.elements = elements;
+    this.recipeIdsModern = recipeIdsModern;
+    this.elementsInit = elementsInit;
+    this.recipeIdsModernInit = recipeIdsModernInit;
+  }
+
+  public WrapperPlayServerUnlockRecipes(Action action, boolean craftingRecipeBookOpen, boolean craftingRecipeBookFilterActive, boolean smeltingRecipeBookOpen,
+                                        boolean smeltingRecipeBookFilterActive, boolean blastFurnaceRecipeBookOpen, boolean blastFurnaceRecipeBookFilterActive,
+                                        boolean smokerRecipeBookOpen, boolean smokerRecipeBookFilterActive, int elements, Optional<String[]> recipeIdsModern,
+                                        OptionalInt elementsInit, Optional<String[]> recipeIdsModernInit) {
+    super(PacketType.Play.Server.UNLOCK_RECIPES);
+    this.action = action;
+    this.craftingRecipeBookOpen = craftingRecipeBookOpen;
+    this.craftingRecipeBookFilterActive = craftingRecipeBookFilterActive;
+    this.smeltingRecipeBookOpen = smeltingRecipeBookOpen;
+    this.smeltingRecipeBookFilterActive = smeltingRecipeBookFilterActive;
+    this.blastFurnaceRecipeBookOpen = blastFurnaceRecipeBookOpen;
+    this.blastFurnaceRecipeBookFilterActive = blastFurnaceRecipeBookFilterActive;
+    this.smokerRecipeBookOpen = smokerRecipeBookOpen;
+    this.smokerRecipeBookFilterActive = smokerRecipeBookFilterActive;
+    this.elements = elements;
+    this.recipeIdsModern = recipeIdsModern;
+    this.elementsInit = elementsInit;
+    this.recipeIdsModernInit = recipeIdsModernInit;
+  }
+
+  @Override
+  public void read() {
+    this.recipeIdsLegacy = Optional.empty();
+    this.recipeIdsModern = Optional.empty();
+    this.elementsInit = OptionalInt.empty();
+    this.recipeIdsLegacyInit = Optional.empty();
+    this.recipeIdsModernInit = Optional.empty();
+
+    if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_12)) {
+      this.action = Action.VALUES[readVarInt()];
+    } else {
+      this.action = Action.VALUES[readShort()];
+    }
+    this.craftingRecipeBookOpen = readBoolean();
+    this.craftingRecipeBookFilterActive = readBoolean();
+    if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_13)) {
+      this.smeltingRecipeBookOpen = readBoolean();
+      this.smeltingRecipeBookFilterActive = readBoolean();
+    }
+    if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_14)) {
```

Comment:
```suggestion
    if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_16_2)) {
```
These properties were added in 1.16.2, not 1.14